lib/app-routes/auth/keys.js

```diff
@@ -81,8 +89,24 @@ exports.register = function(server, options, next) {
                     .then(getHawkCredentials)
                     .then(serveHawkCredentials)
                     .catch((err) => {
+                        
                         server.log(['error', 'login'], err);
                         reply(boom.wrap(err));
+                    })
+                    .catch((err)=>{
+                        reply(boom.wrap(err));
+                    })
+                    .catch(()=>{
```

Comment:
This code block will never be executed, because the `catch` statement above it will catch all errors. You should combine this logic into the statement above.

If this is confusing, let me know, or read up on Promises a bit more. You can also do the nodeschool on Promises to get a better understanding.